test/testapps/BasicTestApp/InteropComponent.cshtml

```diff
@@ -0,0 +1,89 @@
+@using Microsoft.AspNetCore.Blazor.Browser.Interop
+@using BasicTestApp.InteropTest
+@using Microsoft.AspNetCore.Blazor
+
+<button id="btn-interop" onclick="@InvokeInteropAsync">Invoke interop!</button>
+
+<div>
+    <h1>Invocations</h1>
+    @foreach (var invocation in Invocations)
+    {
+        <h2>@invocation.Key</h2>
+        <p id="@invocation.Key">@invocation.Value</p>
+    }
+</div>
+
+<div>
+    <h1>Return values and exceptions thrown from .NET</h1>
+    @foreach (var returnValue in ReturnValues)
+    {
+        <h2>@returnValue.Key</h2>
+        <p id="@returnValue.Key">@returnValue.Value</p>
+    }
+</div>
+
+<div>
+    <h1>Exceptions thrown from JavaScript</h1>
```

Comment:
Are any of these cases observed by the unit tests? The `CanInvokeDotNetMethods` test doesn't seem to read them. Or am I missing where this is used?

Reply:
Not yet, I'm not able to run the E2E tests from my machine so I've been running the tests manually on the browser.

Reply:
This is now covered.